plugin/trino-vertica/src/test/java/io/trino/plugin/vertica/TestVerticaTypeMapping.java

```diff
@@ -0,0 +1,553 @@
+                .addRoundTrip(dateDataType(), LocalDate.of(1970, 1, 1))
+                .addRoundTrip(dateDataType(), LocalDate.of(1970, 2, 3))
+                .addRoundTrip(dateDataType(), LocalDate.of(2017, 7, 1)) // summer on northern hemisphere (possible DST)
+                .addRoundTrip(dateDataType(), LocalDate.of(2017, 1, 1)) // winter on northern hemisphere (possible DST on southern hemisphere)
+                .addRoundTrip(dateDataType(), dateOfLocalTimeChangeForwardAtMidnightInJvmZone)
+                .addRoundTrip(dateDataType(), dateOfLocalTimeChangeForwardAtMidnightInSomeZone)
+                .addRoundTrip(dateDataType(), dateOfLocalTimeChangeBackwardAtMidnightInSomeZone);
+
+        for (String timeZoneId : ImmutableList.of(UTC_KEY.getId(), jvmZone.getId(), someZone.getId())) {
+            Session session = Session.builder(getSession())
+                    .setTimeZoneKey(TimeZoneKey.getTimeZoneKey(timeZoneId))
+                    .build();
+            testCases.execute(getQueryRunner(), session, verticaCreateAndInsert("tpch.test_date"));
+            testCases.execute(getQueryRunner(), session, trinoCreateAsSelect(session, "test_date"));
+            testCases.execute(getQueryRunner(), session, trinoCreateAsSelect(getSession(), "test_date"));
+            testCases.execute(getQueryRunner(), session, trinoCreateAndInsert(session, "test_date"));
+        }
+    }
+
+    @Test
+    public void testReal()
+    {
+        SqlDataTypeTest.create()
+                .addRoundTrip("real", "NULL", DOUBLE, "CAST(NULL AS double)")
+                .addRoundTrip("real", "3.14", DOUBLE, "DOUBLE '3.14'")
+                .addRoundTrip("real", "3.1415927", DOUBLE, "DOUBLE '3.1415927'")
+                .addRoundTrip("real", "'NaN'::real", DOUBLE, "CAST(nan() AS double)")
+                .addRoundTrip("real", "'-Infinity'::real", DOUBLE, "CAST(-infinity() AS double)")
+                .addRoundTrip("real", "'+Infinity'::real", DOUBLE, "CAST(+infinity() AS double)")
+                .execute(getQueryRunner(), verticaCreateAndInsert("tpch.vertica_test_real"));
+
+        SqlDataTypeTest.create()
+                .addRoundTrip("real", "NULL", DOUBLE, "CAST(NULL AS double)")
+                .addRoundTrip("real", "3.14", DOUBLE, "DOUBLE '3.14'")
+                .addRoundTrip("real", "3.1415927", DOUBLE, "DOUBLE '3.1415927'")
+                .addRoundTrip("real", "nan()", DOUBLE, "CAST(nan() AS double)")
+                .addRoundTrip("real", "-infinity()", DOUBLE, "CAST(-infinity() AS double)")
+                .addRoundTrip("real", "+infinity()", DOUBLE, "CAST(+infinity() AS double)")
+                .execute(getQueryRunner(), trinoCreateAsSelect("presto_test_real"));
+    }
+
+    @Test
+    public void testDouble()
+    {
+        SqlDataTypeTest.create()
+                .addRoundTrip("double precision", "NULL", DOUBLE, "CAST(NULL AS double)")
+                .addRoundTrip("double precision", "1.0E100", DOUBLE, "1.0E100")
```

Comment:
Let's test a fractional value too. (Many other connectors are missing this too).